code/modules/projectiles/guns/projectile.dm

```diff
@@ -19,6 +19,7 @@
 	var/obj/item/ammo_storage/magazine/stored_magazine = null
 	var/obj/item/ammo_casing/chambered = null
 	var/mag_type = ""
+	var/obj/item/ammo_storage/magazine/mag_type_restricted = null //better magazine manipulation
```

Comment:
any reason you would not make this a list of types?

Reply:
also, this should be `var/mag_type_restricted`

Reply:
it's a pretty niche variable overall, I don't think it'll see a lot of usage